Content.Client/UserInterface/Controls/RadialContainer.cs

```diff
@@ -68,8 +68,22 @@ public RadialContainer()
 
     }
 
+    private void UpdateRadius()
+    {
+        const float baseRadius = 100f;
+        const float radiusIncrement = 5f;
+
+        var children = ReserveSpaceForHiddenChildren ? Children : Children.Where(x => x.Visible);
+        var childCount = children.Count();
+
+
+        Radius = baseRadius + (childCount * radiusIncrement);
+    }
+
     protected override void Draw(DrawingHandleScreen handle)
     {
+		UpdateRadius();
+		
         var children = ReserveSpaceForHiddenChildren ? Children : Children.Where(x => x.Visible);
         var childCount = children.Count();
 
```

Comment:
You're getting the child count twice and iterating the entire collection, you should be passing this into the method.

You should also document this code.